book_source/03_topical_pages/03_pecan_xml.Rmd

```diff
@@ -768,14 +828,117 @@ The following tags can be used for state data assimilation. More detailed inform
+
+```xml
+<state.data.assimilation>
+  <state.variables>
+   <variable>
+    <variable.name>AbvGrndWood</variable.name>
+    <unit>MgC/ha</unit>
+    <min_value>0</min_value>
+    <max_value>9999</max_value>
+   </variable>
+   <variable>
+    <variable.name>LAI</variable.name>
+    <unit></unit>
+    <min_value>0</min_value>
+    <max_value>9999</max_value>
+   </variable>
+   <variable>
+  <variable.name>SoilMoistFrac</variable.name>
+  <unit></unit>
+  <min_value>0</min_value>
+  <max_value>100</max_value>
+  </variable>
+   <variable>
+    <variable.name>TotSoilCarb</variable.name>
+    <unit>kg/m^2</unit>
+    <min_value>0</min_value>
+    <max_value>9999</max_value>
+   </variable>
+  </state.variables>
+</state.data.assimilation>
+```
+
+* **variable** : [required] Each of this section should include each state variables separately.
+* **variable.name** : [required] The name of the state variable (same as what we call it in the model output file. for example, "SIPNET.out" file).
+* **unit** : [required] The unit of the state variable (can be empty if it's unitless).
+* **min_value** : [required] The minimum range of the state variable.
+* **max_value** : [required] The maximum range of the state variable.
+
+### Observation Preparation for State Data Assimilation {#xml-observation-preparation}
+
+The following tags can be used for observation preparation for the SDA workflow.
+
+```xml
+<state.data.assimilation>
+  <Obs_Prep>
+   <Landtrendr_AGB>
+    <AGB_input_dir>/projectnb/dietzelab/dongchen/Multi-site/download_500_sites/AGB</AGB_input_dir>
+    <allow_download>TRUE</allow_download>
+    <export_csv>TRUE</export_csv>
+    <timestep>
+     <unit>year</unit>
+     <num>1</num>
+    </timestep>
+   </Landtrendr_AGB>
+   
+   <MODIS_LAI>
+    <search_window>30</search_window>
+    <export_csv>TRUE</export_csv>
+    <run_parallel>TRUE</run_parallel>
+    <timestep>
+     <unit>year</unit>
+     <num>1</num>
+    </timestep>
+   </MODIS_LAI>
+   
+   <SMAP_SMP>
+    <search_window>30</search_window>
+    <export_csv>TRUE</export_csv>
+    <update_csv>FALSE</update_csv>
+    <timestep>
+     <unit>year</unit>
+     <num>1</num>
+    </timestep>
+   </SMAP_SMP>
+   
+   <Soilgrids_SoilC>
+    <timestep>
+     <unit>year</unit>
+     <num>1</num>
+    </timestep>
+   </Soilgrids_SoilC>
+   <outdir>/projectnb/dietzelab/dongchen/All_NEON_SDA/test_OBS</outdir>
+   <start.date>2012-07-15</start.date>
+   <end.date>2021-07-15</end.date>
+  </Obs_Prep>
+</state.data.assimilation>
+```
+
+* **Landtrendr_AGB** : [optional] The naming convention of this section is based on "DataSource_AbbreviationOfData", if you want to proceed with the SDA_Obs_Assembler function and generate obs.mean and obs.cov of the landtrendr AGB data, you need to specify it in a similar style.
+* **AGB_input_dir** : [optional] This tag is only for Landtrendr_AGB_prep function, which points to the directory where the pre-downloaded AGB files existed.
```

Comment:
I'd change it to AGB_indir, the reason that I use AGB at the beginning is to avoid confusion and differentiate data source from other potential future observations.